### ISSUE TITLE
(Optionally) open share dialog by default

### ISSUE THREAD

**Issue by szpak** [Tommy-Geenexus/exif-eraser#4]
I use exif-eraser mostly to share photo from a gallery or a camera app to remove exif and immediately send it via MMS or some social network. In the past you implemented (for me ;) ) a very useful feature to automatically ask about further sharing saved image with the other apps (MMS, Twitter, etc.).

In the new version, it is still possible by using the "share icon", however, for people using exif-erase only (or mostly) in the way I described, it is "one more click" to finish the operation. What do you think about adding an extra settings switch (disabled by default) to automatically open the "sharing dialog" after a stripped image is saved? In a case I don't want to share the image immediately, I could just tap "back" and close the dialog.

Simple steps:
1. Gallery -> Share with -> exif-eraser
2. Confirm place and name to save stripped image
3. Automatically see "share with" dialog open (without clicking the "share with" icon in exif-eraser).

**Comment by Tommy-Geenexus** [comment]
In a vacuum the number of steps it takes to complete sharing hasn't changed since v2.0.

Where before it was "save image" -> "confirm share dialog", it now is "save image" -> "share via menu", both consuming 2 clicks.
The positioning may be less convenient, but this is to maintain layout scalability down to tiny DPI screens.
If the menu ever was to be expanded, there would be more steps involved.

The irony is i always considered the share dialog bad UX and wanted it gone,
and now that it's gone someone actually found it to be useful.

TL;DR I will not implement a dialog as of now, I don't think it's worth the effort .

**Comment by szpak** [comment]
In fact, I don't need an extra dialog in exif-eraser itself. I would like to have the share button in your app "clicked" automatically (triggering the **native sharing dialog from Android** where is can select the app to share with) if the aforementioned switch in the settings enabled (for people removing exif metadata to immediately share the output picture with some other app).
Tapping back once should close that (Android native) dialog (in a situation that I changed my mind or this time I wanted to just remove exif metadata for some reason). Would it be more intuitive in your opinion?

**Comment by Tommy-Geenexus** [comment]
Sure, this could be implemented without much effort.

**Comment by szpak** [comment]
I've just tested it with 3.0.2 and it works as expected, thanks!

Btw, I wonder, if it would be sensible (and not overly complicated) to detect that exif-eraser was triggered by the "share with" action (instead of being started using the application icon) and just close/hide the application window on tapping "back" on the "summary page" (instead of showing the main application window - that with "choose your preferred  image source").

Currently (especially with "automatic sharing" enabled) it is required to tap back once more after sharing stripped image with some other app to close exif-eraser (and come back to the app that triggered the "share with" action). WDYT?

**Comment by Tommy-Geenexus** [comment]
It would be simple to detect, but i think it would go too far.
While this change would benefit your usecase, it probably would confuse lots of users, so I don't think it makes sense to change upwards navigation handling.

Also, you're better off upgrading to 3.0.3, it contains bugfixes related to the share by default feature.

**Comment by szpak** [comment]
I will. It was not available awhile ago in Google Play.

Regarding the original proposed behavior, as you wish. It might be somehow confusing for newcomers. However, the majority of people calling "send with" from the other app don't expect to see "choose your preferred image source". They usually share with the app to get stripped images. Otherwise they would start the app using the icon.

Btw, yesterday I had an opportunity to share multiple images with exif-eraser to share them with the app. I really like that in 3.x it is possible to do it in one step!

**Comment by Tommy-Geenexus** [comment]
> However, the majority of people calling "send with" from the other app don't expect to see "choose your preferred image source". They usually share with the app to get stripped images. Otherwise they would start the app using the icon.

The system in place (the jetpack navigation component) deeplinks the user directly to the image ui component, while creating a proper backstack to handle upwards navigation.
The navigation flow atm is like it "should" be, as defined by the android navigation component and i think it makes sense this way. If i deeplink into the app i still expect to have full functionality if needed.

> Btw, yesterday I had an opportunity to share multiple images with exif-eraser to share them with the app. I really like that in 3.x it is possible to do it in one step!

The UX in previous versions was pretty bad, right now a more sane approach is in place.
Ideally there would be a bottomsheet containing scaled images which could be selected for sharing instead of defaulting to sharing everything, but as far as i know this is currently not possible due to the permissionless approach and/or lack of support in third party libraries.
